packages/apps/graph/src/graph/query/fungible-accounts-by-public-key.ts

```diff
@@ -0,0 +1,109 @@
+import { prismaClient } from '@db/prisma-client';
+import { getFungibleChainAccount } from '@services/account-service';
+import { chainIds } from '@utils/chains';
+import { normalizeError } from '@utils/errors';
+import { builder } from '../builder';
+import FungibleAccount from '../objects/fungible-account';
+import type { FungibleChainAccount } from '../types/graphql-types';
+import { FungibleAccountName } from '../types/graphql-types';
+
+builder.queryField('fungibleAccountsByPublicKey', (t) =>
+  t.field({
+    description: 'Retrieve an account by public key.',
+    args: {
+      publicKey: t.arg.string({
+        required: true,
+        validate: {
+          minLength: 1,
+        },
+      }),
+      fungibleName: t.arg.string({
+        required: false,
+        validate: {
+          minLength: 1,
+        },
+      }),
+    },
+    type: [FungibleAccount],
+    nullable: true,
+    async resolve(__parent, args) {
+      try {
+        const accountAndFungibleNames =
+          await getAccountAndFungibleNamesByPublicKey(
+            args.publicKey,
+            args.fungibleName ?? undefined,
+          );
+
+        if (accountAndFungibleNames.length === 0) {
+          return null;
+        }
+
+        const fungibleAccounts = await Promise.all(
+          accountAndFungibleNames.map(async (accountAndFungibleName) => {
+            const chainAccounts = (
+              await Promise.all(
+                chainIds.map((chainId) =>
+                  getFungibleChainAccount({
+                    chainId: chainId,
+                    fungibleName: accountAndFungibleName.fungiblename,
+                    accountName: accountAndFungibleName.accountname,
+                  }),
+                ),
+              )
+            ).filter(Boolean) as FungibleChainAccount[];
+
+            return {
+              __typename: FungibleAccountName,
+              accountName: accountAndFungibleName.accountname,
+              fungibleName: accountAndFungibleName.fungiblename,
+              chainAccounts,
+              totalBalance: 0,
+              transactions: [],
+              transfers: [],
+            };
+          }),
+        );
+
+        return fungibleAccounts;
+      } catch (error) {
+        throw normalizeError(error);
+      }
+    },
+  }),
+);
+
+async function getAccountAndFungibleNamesByPublicKey(
+  publicKey: string,
+  fungible?: string,
+): Promise<{ accountname: string; fungiblename: string }[]> {
+  const regex = /^[a-zA-Z0-9]+$/;
+
+  if (publicKey.length !== 64 || !regex.test(publicKey)) {
+    throw new Error('Invalid public key');
+  }
+
+  // This is required because prisma queryRaw does not support string interpolation for entire query fragments.
+  if (fungible) {
```

Comment:
Is there a reason we use raw queries here? Distinct selection is possible with Prisma: https://www.prisma.io/docs/orm/prisma-client/queries/aggregation-grouping-summarizing#select-distinct

Reply:
We are using a raw query because it allows us to look for a `publicKey` in the transaction field `data::text`. Since this is a JSON field, when not using raw query we are obliged to provide a path. This path is dynamic and can be different from transaction to transaction